neqo-qpack/src/encoder.rs

```diff
@@ -211,50 +211,49 @@ impl QPackEncoder {
 
     /// Inserts a new entry into a table and sends the corresponding instruction to a peer. An entry is added only
     /// if it is possible to send the corresponding instruction immediately, i.e. the encoder stream is not
-    /// blocked by the flow control.
+    /// blocked by the flow control (or stream internal buffer(this is very unlikely)).
     /// ### Errors
     /// `EncoderStreamBlocked` if the encoder stream is blocked by the flow control.
     /// The function can return transport errors.
-    /// `HeaderLookup` if `InsertWithNameRefStatic`, `InsertWithNameRefDynamic` or `Duplicate` is used and the
-    /// reference entry cannot be found.
     /// `DynamicTableFull` if the dynamic table does not have enough space for the entry.
-    pub fn insert(&mut self, conn: &mut Connection, instruction: &EncoderInstruction) -> Res<u64> {
-        qdebug!([self], "insert instruction {:?}.", instruction);
+    pub fn insert(&mut self, conn: &mut Connection, name: &[u8], value: &[u8]) -> Res<u64> {
+        qdebug!([self], "insert {:?} {:?}.", name, value);
         self.send(conn)?;
         if self.send_buf.len() != 0 {
             return Err(Error::EncoderStreamBlocked);
         }
 
+        let entry_size =
+            u64::try_from(name.len() + value.len() + ADDITIONAL_TABLE_ENTRY_SIZE).unwrap();
+
+        if !self.table.check_insert_possible(entry_size) {
```

Comment:
This is another case where usize is probably a better type for the argument to check_insert_possible (I would call it just `insert_possible`).